src/heightmap.cpp

```diff
@@ -11,6 +11,70 @@
 
 #include <queue>
 
+
+MapLoader::MapLoader()
+{
+    ros::SubscribeOptions map_sub_options = ros::SubscribeOptions::create<nav_msgs::OccupancyGrid>("/map", 1, boost::bind(&MapLoader::mapCallback, this, _1), ros::VoidPtr(), &cb_map);
+    sub_map = nh.subscribe(map_sub_options);
+}
+
+void MapLoader::getMap(nav_msgs::OccupancyGrid& mapRef)
+{
+    if (!initialized)
+    {
+        load();
+    }
+    mapRef = map;
+}
+
+void MapLoader::getMapMetadata(nav_msgs::MapMetaData& metadataRef)
+{
+    if (!initialized)
+    {
+        load();
+    }
+    metadataRef = map.info;
+}
+
+void MapLoader::getMapImage(cv::Mat& imageRef)
+{
+    if (!initialized)
+    {
+        load();
+    }
+    imageRef = mapImage;
+}
+
+void MapLoader::load()
+{
+    ROS_INFO_STREAM("Waiting for map");
+    while(!initialized)
+    {
+        cb_map.callAvailable();
+    }
```

Comment:
This is done at an unlimited rate...

Reply:
Yes, this requires a rospy.rate probably running this at 10 hz is enough